src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -0,0 +1,167 @@
+        upStationSection.ifPresent(section -> section.updateUpStation(newSection));
+        downStationSection.ifPresent(section -> section.updateDownStation(newSection));
+        sections.add(newSection);
+    }
+
+    private void validateDuplicateSection(Section section) {
+        if(isAllContainStations(section)) {
+            throw new IllegalArgumentException(ErrorCode.이미_존재하는_구간.getErrorMessage());
+        }
+    }
+
+    private boolean isAllContainStations(Section section) {
+        return findStations().containsAll(section.stations());
+    }
+
+    private void validateNotContainAnySection(Section section) {
+        if(isNotContainAnyStation(section)) {
+            throw new IllegalArgumentException(ErrorCode.구간의_상행역과_하행역이_모두_노선에_포함되지_않음.getErrorMessage());
+        }
+    }
+
+    private boolean isNotContainAnyStation(Section section) {
+        return findStations().stream()
+                .noneMatch(station -> section.stations().contains(station));
+    }
+
+    private void validateLine(Section section) {
+        if(isNotEqualLine(section)) {
+            throw new IllegalArgumentException(ErrorCode.구간의_노선이_기존_구간들과_상이함.getErrorMessage());
+        }
+    }
+
+    private boolean isNotEqualLine(Section section) {
+        if(sections.isEmpty()) {
+            return false;
+        }
+        return !sections.get(0).hasSameLine(section);
+    }
+
+    public void removeStationInLine(Station deleteStation) {
+        validateIfOnlyOneSection();
+        Optional<Section> upStationSection = findSectionByUpStation(deleteStation);
+        Optional<Section> downStationSection = findSectionByDownStation(deleteStation);
+        validateNotContainAnySection(upStationSection.isPresent(), downStationSection.isPresent());
+
+        if(isInTheMiddleOfLine(upStationSection.isPresent(), downStationSection.isPresent())) {
+            addCombineSection(downStationSection.get(), upStationSection.get());
+        }
+        upStationSection.ifPresent(this::deleteSection);
+        downStationSection.ifPresent(this::deleteSection);
+    }
+
+    private void validateIfOnlyOneSection() {
+        if(sections.size() == ONLY_ONE_SECTION) {
+            throw new IllegalArgumentException(ErrorCode.노선에_속한_구간이_하나이면_제거_불가.getErrorMessage());
+        }
+    }
+
+    private Optional<Section> findSectionByUpStation(Station station) {
+        return sections.stream()
+                .filter(section -> section.isSameUpStation(station))
+                .findFirst();
+    }
+
+    private Optional<Section> findSectionByDownStation(Station station) {
+        return sections.stream()
+                .filter(section -> section.isSameDownStation(station))
+                .findFirst();
+    }
+
+    private void validateNotContainAnySection(boolean hasUpStationSection, boolean hasDownStationSection) {
+        if(hasNotBothUpAnddownStationSection(hasUpStationSection, hasDownStationSection)) {
```

Comment:
```suggestion
        if(hasNotBothUpAndDownStationSection(hasUpStationSection, hasDownStationSection)) {
```
중요하진 않습니다만... 보여서 체크해둡니다 ㅎ.ㅎ;;